### ISSUE TITLE
ClosestPoint() implementation for meshes

### ISSUE THREAD

**Issue by iltabe** [GSharker/G-Shark#416]
### Discussed in https://github.com/GSharker/G-Shark/discussions/415

<div type='discussions-op-text'>

<sup>Originally posted by **iltabe** May 30, 2023</sup>
Hi guys!

I know some of you for different reasons! Ciao!

For a client I'm building an algorithm that requires some geometrical operations. I've looked into G-Shark and started to use it. It looks promising and does almost everything I need (basic vector, point, plane stuff).
Thanks guys for having give birth to this nice project!

The only thing I miss is a `mesh.ClosestPoint(Point3 pt)` method... Is it in the todo list by any luck?
Thanks!
Gianluca</div>

It would be nice to implement some functionality for meshes, especially `myMesh.ClosestPoint(Point3 test_point)`.
The behaviour I expect is to get back from the function the real closest point:

1. on a mesh vertex, if the mesh vertex is the closest point to the test point
2. on a mesh face boundary (edge) if that is the closest point to the test point
3. a point inside a mesh face if the length of the line perpendicular to the face (and passing per test_point) is shorter to any other line connecting test_point to the mesh.

Long story short: the closest point! :)

**Comment by iltabe** [comment]
Hi there!

We found a bit of internal resources to work on this, so we started a first implementation.
We will open a PR in the next few days/weeks. 
Thanks!

**Comment by d3ssy** [comment]
Excellent! Looking forward to it. Thank you for taking the time to
contribute.

**Comment by iltabe** [comment]
Ciao @d3ssy,
The PR regarding mesh closest point is out. Who is the best person to be tagged as reviewer?
Thanks!

**Comment by d3ssy** [comment]
I'll review later today, cheers!

**Comment by iltabe** [comment]
Cool, thank you!

Il giorno mer 26 lug 2023 alle ore 10:47 Guido Maciocci <
***@***.***> ha scritto:

**Comment by d3ssy** [comment]
@iltabe Review done! Apologies for the delay. Please reach out if you need any help or further info!
